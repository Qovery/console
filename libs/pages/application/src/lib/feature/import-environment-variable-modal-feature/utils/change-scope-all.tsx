import { APIVariableScopeEnum } from 'qovery-typescript-axios'
import { UseFormSetValue } from 'react-hook-form'

export function changeScopeForAll(
  value: APIVariableScopeEnum,
  setValue: UseFormSetValue<{ [key: string]: string }>,
  keys: (string | boolean)[]
): void {
  keys.forEach((key) => {
    setValue(key + '_scope', value?.toString())
  })
}
