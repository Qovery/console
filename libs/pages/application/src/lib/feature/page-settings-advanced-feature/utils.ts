import { ApplicationAdvancedSettings } from 'qovery-typescript-axios'
import { ApplicationEntity } from '@console/shared/interfaces'

export function initFormValues(keys: string[], application: ApplicationEntity): { [key: string]: string } {
  const values: { [key: string]: string } = {}
  keys.forEach((key) => {
    if (application.advanced_settings?.current_settings) {
      values[key] =
        application.advanced_settings.current_settings[key as keyof ApplicationAdvancedSettings]?.toString() || ''
    }
  })

  return values
}
