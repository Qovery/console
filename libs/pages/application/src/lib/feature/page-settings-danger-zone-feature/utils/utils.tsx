import { UseFormRegister, UseFormUnregister } from 'react-hook-form/dist/types/form'

export const onAddStorage = (
  register: UseFormRegister<{ [key: string]: string | number }>,
  keys: string[]
): string[] => {
  const id = Math.trunc(Math.random() * 10000).toString()

  register('size_' + id, { value: '' })
  register('type_' + id, { value: '' })
  register('path_' + id, { value: '' })

  return [...keys, id]
}

export const onRemove = (
  key: string,
  unregister: UseFormUnregister<{ [key: string]: string | number }>,
  keys: string[]
): string[] => {
  unregister('size_' + key)
  unregister('type_' + key)
  unregister('path_' + key)

  return keys.filter((k) => k !== key)
}
