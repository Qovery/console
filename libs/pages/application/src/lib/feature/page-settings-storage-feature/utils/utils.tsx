import { UseFormRegister, UseFormUnregister } from 'react-hook-form/dist/types/form'
import { ApplicationStorageStorage } from 'qovery-typescript-axios'

export const addStorage = (register: UseFormRegister<{ [key: string]: string | number }>, keys: string[]): string[] => {
  const id = Math.trunc(Math.random() * 10000).toString()

  register('size_' + id, { value: '' })
  register('type_' + id, { value: '' })
  register('path_' + id, { value: '' })

  return [...keys, id]
}

export const onRemove = (
  key: string,
  unregister: UseFormUnregister<{ [key: string]: string | number }>,
  keys: string[]
): string[] => {
  unregister('size_' + key)
  unregister('type_' + key)
  unregister('path_' + key)

  return keys.filter((k) => k !== key)
}

export const initStorage = (
  register: UseFormRegister<{ [key: string]: string | number }>,
  storages: ApplicationStorageStorage[]
): string[] => {
  const keys: string[] = []

  storages.forEach((storage) => {
    register('size_' + storage.id, { value: storage.size })
    register('type_' + storage.id, { value: storage.type })
    register('path_' + storage.id, { value: storage.mount_point })

    keys.push(storage.id || '')
  })

  return keys
}
