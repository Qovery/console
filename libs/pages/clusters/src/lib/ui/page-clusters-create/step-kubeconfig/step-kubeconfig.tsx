import { type FormEventHandler, useCallback } from 'react'
import { useDropzone } from 'react-dropzone'
import { useController, useFormContext } from 'react-hook-form'
import { useNavigate, useParams } from 'react-router-dom'
import { type ClusterKubeconfigData } from '@qovery/shared/interfaces'
import { CLUSTERS_CREATION_GENERAL_URL, CLUSTERS_CREATION_URL, CLUSTERS_URL } from '@qovery/shared/routes'
import { Button, Dropzone, Heading, Icon, Section } from '@qovery/shared/ui'

export interface StepKubeconfigProps {
  onSubmit: FormEventHandler<HTMLFormElement>
}

export function StepKubeconfig({ onSubmit }: StepKubeconfigProps) {
  const navigate = useNavigate()
  const { organizationId = '' } = useParams()
  const { control, formState } = useFormContext<ClusterKubeconfigData>()
  const { field: fileName } = useController<ClusterKubeconfigData>({
    name: 'file_name',
    control,
    rules: { required: true },
  })
  const { field: fileContent } = useController<ClusterKubeconfigData>({
    name: 'file_content',
    control,
    rules: { required: true },
  })
  const { field: fileSize } = useController<ClusterKubeconfigData>({
    name: 'file_size',
    control,
    rules: { required: true },
  })

  const onDrop = useCallback((acceptedFiles: File[]) => {
    acceptedFiles.forEach((file) => {
      const reader = new FileReader()

      reader.onabort = () => console.log('file reading was aborted')
      reader.onerror = () => console.log('file reading has failed')
      reader.onload = () => {
        fileSize.onChange(file.size / 1000)
        fileName.onChange(file.name)
        fileContent.onChange(reader.result)
      }
      reader.readAsText(file)
    })
  }, [])
  const { getRootProps, getInputProps, isDragActive } = useDropzone({
    onDrop,
    maxFiles: 1,
  })

  const handleDelete = () => {
    fileName.onChange(null)
    fileContent.onChange(null)
  }

  return (
    <Section>
      <div className="mb-10">
        <Heading className="mb-2">Kubeconfig</Heading>
        <p className="text-neutral-350">Upload your Kubeconfig file here.</p>
      </div>
      <form onSubmit={onSubmit}>
        <div className="mb-10">
          {fileName.value ? (
            <div className="flex items-center gap-2 rounded border border-neutral-200 p-3">
              <div className="p-2">
                <Icon iconName="file-lines" />
              </div>
              <div className="flex grow flex-col text-xs">
                <span className="font-medium text-neutral-400">{fileName.value}</span>
                <span className="text-neutral-350">{fileSize.value} Ko</span>
              </div>
              <div>
                <Button type="button" variant="outline" color="neutral" onClick={handleDelete}>
                  <Icon iconName="trash" />
                </Button>
              </div>
            </div>
          ) : (
            <div {...getRootProps({ className: 'dropzone' })}>
              <input data-testid="drop-input" {...getInputProps()} />
              <Dropzone isDragActive={isDragActive} typeFile="kubeconfig" />
            </div>
          )}
        </div>
        <div className="flex justify-between">
          <Button
            size="lg"
            type="button"
            variant="surface"
            color="neutral"
            onClick={() =>
              navigate(`${CLUSTERS_URL(organizationId)}${CLUSTERS_CREATION_URL}${CLUSTERS_CREATION_GENERAL_URL}`)
            }
          >
            Back
          </Button>
          <Button size="lg" data-testid="button-submit" type="submit" disabled={!formState.isValid}>
            Continue
          </Button>
        </div>
      </form>
    </Section>
  )
}

export default StepKubeconfig
