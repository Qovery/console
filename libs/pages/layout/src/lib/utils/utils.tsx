export function setCurrentOrganizationIdOnStorage(organizationId: string) {
  localStorage.setItem('currentOrganizationId', organizationId)
}

export function setCurrentProjectIdOnStorage(projectId: string) {
  localStorage.setItem('currentProjectId', projectId)
}
