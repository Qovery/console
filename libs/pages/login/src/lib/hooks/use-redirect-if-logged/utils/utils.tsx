export function getRedirectLoginUriFromStorage(): string | null {
  return localStorage.getItem('redirectLoginUri')
}

export function getCurrentOrganizationIdFromStorage(): string | null {
  return localStorage.getItem('currentOrganizationId')
}

export function getCurrentProjectIdFromStorage(): string | null {
  return localStorage.getItem('currentProjectId')
}
