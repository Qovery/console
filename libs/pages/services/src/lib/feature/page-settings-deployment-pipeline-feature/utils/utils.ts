import { type DeploymentStageResponse, type DeploymentStageServiceResponse } from 'qovery-typescript-axios'
import { type DraggableLocation } from 'react-beautiful-dnd'

export const reorderService = (
  stages: DeploymentStageResponse[],
  destinationIndex: number,
  startIndex: number,
  endIndex: number
): DeploymentStageResponse[] => {
  const cloneStages = [...stages]
  const currentStage = cloneStages[destinationIndex]
  const currentStageServices = [...(currentStage.services as DeploymentStageServiceResponse[])]

  const [removed] = currentStageServices.splice(startIndex, 1)
  currentStageServices.splice(endIndex, 0, removed)

  // add reorder on the clone stages array
  const result = cloneStages.map((stage, index) => {
    if (index === destinationIndex) {
      return { ...stage, services: currentStageServices }
    }
    return stage
  })

  return result
}

export const move = (
  stages: DeploymentStageResponse[],
  droppableSource: DraggableLocation,
  droppableDestination: DraggableLocation
): DeploymentStageResponse[] | undefined => {
  let result
  const cloneStages = [...stages]

  const sourceDroppableId = parseInt(droppableSource.droppableId, 10)
  const destinationDroppableId = parseInt(droppableDestination.droppableId, 10)

  // find the source stage
  const sourceStage = cloneStages.find((stage, index) => index === sourceDroppableId)

  if (sourceStage) {
    // find the destination stage
    const destinationStage = cloneStages.find((stage, index) => index === destinationDroppableId)

    if (destinationStage?.services && sourceStage?.services) {
      const cloneSourceServices = [...sourceStage.services]
      // remove the service from the source services
      const [serviceMoved] = cloneSourceServices.splice(droppableSource.index, 1)

      // add the service on the destination services on the droppable destination index
      const cloneDestinationServices = [...destinationStage.services]
      cloneDestinationServices.splice(droppableDestination.index, 0, serviceMoved)

      // add new services for source and destination on the stages
      result = cloneStages.map((stage, index) => {
        if (index === sourceDroppableId) {
          return { ...stage, services: cloneSourceServices }
        }
        if (index === destinationDroppableId) {
          return { ...stage, services: cloneDestinationServices }
        }
        return stage
      })
    }
  }

  return result
}
