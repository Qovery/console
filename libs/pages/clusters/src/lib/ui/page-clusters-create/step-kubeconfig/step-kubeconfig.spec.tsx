import { wrapWithReactHookForm } from '__tests__/utils/wrap-with-react-hook-form'
import { renderWithProviders } from '@qovery/shared/util-tests'
import { StepKubeconfig } from './step-kubeconfig'

describe('StepKubeconfig', () => {
  it('should match snapshot', () => {
    const onSubmit = jest.fn()
    const { container } = renderWithProviders(
      wrapWithReactHookForm(<StepKubeconfig onSubmit={onSubmit} />, {
        defaultValues: {},
      })
    )
    expect(container).toMatchSnapshot()
  })
  it('should match snapshot with data', () => {
    const onSubmit = jest.fn()
    const { container } = renderWithProviders(
      wrapWithReactHookForm(<StepKubeconfig onSubmit={onSubmit} />, {
        defaultValues: {
          file_name: 'file.yaml',
          file_content: '',
          file_size: 1234,
        },
      })
    )
    expect(container).toMatchSnapshot()
  })
})
