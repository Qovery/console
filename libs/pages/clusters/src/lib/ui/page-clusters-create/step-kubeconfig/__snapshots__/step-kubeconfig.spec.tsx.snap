// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StepKubeconfig should match snapshot 1`] = `
<div>
  <section
    class="flex flex-col"
  >
    <div
      class="mb-10"
    >
      <h1
        class="text-neutral-400 text-xl font-bold mb-2"
      >
        Kubeconfig
      </h1>
      <p
        class="text-neutral-350"
      >
        Upload your Kubeconfig file here.
      </p>
    </div>
    <form>
      <div
        class="mb-10"
      >
        <div
          class="dropzone"
          role="presentation"
          tabindex="0"
        >
          <input
            data-testid="drop-input"
            multiple=""
            style="display: none;"
            tabindex="-1"
            type="file"
          />
          <div
            class="flex min-h-[140px] w-full cursor-pointer items-center justify-center rounded border border-neutral-250 bg-neutral-100 text-xs text-neutral-350 transition-all duration-150 ease-in-out hover:border-neutral-300 border-dashed"
          >
            <div
              class="flex h-full flex-col items-center justify-center text-center"
            >
              <i
                aria-hidden="true"
                class="fa-solid fa-file-arrow-down mb-2 text-xl"
              />
              <p>
                Click to import your 
                kubeconfig
                 file or
                <br />
                drag and drop it
              </p>
            </div>
          </div>
        </div>
      </div>
      <div
        class="flex justify-between"
      >
        <button
          class="inline-flex items-center shrink-0 font-medium transition focus-visible:[&:not(:active)]:outline-2 outline-0 shadow-sm disabled:shadow-none active:shadow-none outline-brand-500 text-sm h-11 px-5 rounded border border-neutral-250 dark:border-neutral-400 text-neutral-400 dark:text-white hover:[&:not(:active)]:border-neutral-300 dark:hover:[&:not(:active):not(:disabled)]:border-neutral-350 active:bg-neutral-150 disabled:text-neutral-300 disabled:bg-neutral-100 disabled:border-none bg-neutral-100 dark:bg-neutral-500 hover:[&:not(:active):not(:disabled)]:bg-white dark:hover:[&:not(:active):not(:disabled)]:bg-neutral-400"
          type="button"
        >
          Back
        </button>
        <button
          class="inline-flex items-center shrink-0 font-medium transition focus-visible:[&:not(:active)]:outline-2 outline-0 shadow-sm disabled:shadow-none active:shadow-none outline-neutral-500 text-sm h-11 px-5 rounded bg-brand-500 active:bg-brand-600 hover:bg-brand-400 text-white disabled:text-brand-300 disabled:bg-brand-100"
          data-testid="button-submit"
          disabled=""
          type="submit"
        >
          Continue
        </button>
      </div>
    </form>
  </section>
</div>
`;

exports[`StepKubeconfig should match snapshot with data 1`] = `
<div>
  <section
    class="flex flex-col"
  >
    <div
      class="mb-10"
    >
      <h1
        class="text-neutral-400 text-xl font-bold mb-2"
      >
        Kubeconfig
      </h1>
      <p
        class="text-neutral-350"
      >
        Upload your Kubeconfig file here.
      </p>
    </div>
    <form>
      <div
        class="mb-10"
      >
        <div
          class="flex items-center gap-2 rounded border border-neutral-200 p-3"
        >
          <div
            class="p-2"
          >
            <i
              aria-hidden="true"
              class="fa-solid fa-file-lines "
            />
          </div>
          <div
            class="flex grow flex-col text-xs"
          >
            <span
              class="font-medium text-neutral-400"
            >
              file.yaml
            </span>
            <span
              class="text-neutral-350"
            >
              1234
               Ko
            </span>
          </div>
          <div>
            <button
              class="inline-flex items-center shrink-0 font-medium transition focus-visible:[&:not(:active)]:outline-2 outline-0 shadow-sm disabled:shadow-none active:shadow-none bg-white dark:bg-neutral-600 outline-brand-500 text-xs h-7 px-2 rounded border border-neutral-250 dark:border-neutral-400 text-neutral-400 dark:text-white hover:[&:not(:active)]:border-neutral-300 dark:hover:[&:not(:active):not(:disabled)]:border-neutral-350 active:bg-neutral-150 disabled:text-neutral-300 disabled:bg-neutral-100 disabled:border-none"
              type="button"
            >
              <i
                aria-hidden="true"
                class="fa-solid fa-trash "
              />
            </button>
          </div>
        </div>
      </div>
      <div
        class="flex justify-between"
      >
        <button
          class="inline-flex items-center shrink-0 font-medium transition focus-visible:[&:not(:active)]:outline-2 outline-0 shadow-sm disabled:shadow-none active:shadow-none outline-brand-500 text-sm h-11 px-5 rounded border border-neutral-250 dark:border-neutral-400 text-neutral-400 dark:text-white hover:[&:not(:active)]:border-neutral-300 dark:hover:[&:not(:active):not(:disabled)]:border-neutral-350 active:bg-neutral-150 disabled:text-neutral-300 disabled:bg-neutral-100 disabled:border-none bg-neutral-100 dark:bg-neutral-500 hover:[&:not(:active):not(:disabled)]:bg-white dark:hover:[&:not(:active):not(:disabled)]:bg-neutral-400"
          type="button"
        >
          Back
        </button>
        <button
          class="inline-flex items-center shrink-0 font-medium transition focus-visible:[&:not(:active)]:outline-2 outline-0 shadow-sm disabled:shadow-none active:shadow-none outline-neutral-500 text-sm h-11 px-5 rounded bg-brand-500 active:bg-brand-600 hover:bg-brand-400 text-white disabled:text-brand-300 disabled:bg-brand-100"
          data-testid="button-submit"
          disabled=""
          type="submit"
        >
          Continue
        </button>
      </div>
    </form>
  </section>
</div>
`;
